amethyst_utils/src/render.rs

```diff
@@ -0,0 +1,58 @@
+//! Module containing utility methods to easily create `Material` and `Texture` handles using a minimal amount of code.
+
+use amethyst_assets::{AssetStorage, Handle, Loader, Progress};
+use amethyst_core::specs::{Read, ReadExpect};
+use amethyst_renderer::{Material, MaterialDefaults, PngFormat, Texture, TextureMetadata};
+
+/// SystemData used to create `Material` instances in various ways.
+///
+/// # Example
+/// ```rust, ignore
+/// let mat = world.exec(|creator: MaterialCreator|
+///     creator.material_from_color([0.0, 1.0, 0.0, 1.0], ())
+/// );
+/// ```
+#[derive(SystemData)]
+pub struct MaterialCreator<'a> {
+    loader: ReadExpect<'a, Loader>,
+    storage: Read<'a, AssetStorage<Texture>>,
+    defaults: ReadExpect<'a, MaterialDefaults>,
+}
+
+impl<'a> MaterialCreator<'a> {
+    /// Generate a `Material` from a color.
+    pub fn material_from_color<T: Progress>(
+        &self,
+        color: [f32; 4],
+        progress_counter: T,
+    ) -> Material {
+        let albedo = self
+            .loader
+            .load_from_data(color.into(), progress_counter, &self.storage);
+        self.material_from_texture(albedo)
+    }
+
+    /// Generate a `Material` from a filesystem path pointing to a png image.
+    /// Uses the "resource path" provided at Amethyst's initialization.
+    pub fn material_from_png<T: Progress, S: Into<String>>(
+        &self,
+        path: S,
+        progress_counter: T,
+    ) -> Material {
+        self.material_from_texture(self.loader.load(
+            path,
+            PngFormat,
+            TextureMetadata::srgb(),
+            progress_counter,
+            &self.storage,
+        ))
+    }
+
+    /// Generate a `Material` from a texture handle.
+    pub fn material_from_texture(&self, texture: Handle<Texture>) -> Material {
```

Comment:
I'm not sure how useful this actually is at the moment, it seems like just doing the contained code whenever you want a texture with an albedo is pretty simple... perhaps we could make this https://github.com/amethyst/amethyst/blob/master/amethyst_renderer/src/pass/util.rs#L24 a public part of the `mtl` module and then make this method `material_from_textures` something like:

```rs
pub fn material_from_textures(&self, textures: &[(TextureType, Handle<Texture>)] -> Material {
    ...
}
```
Which inserts the given textures into the correct slots and then uses default for the rest?